examples/scripts/spm_CMAES.py

```diff
@@ -33,7 +33,7 @@
 ]
 
 # Generate problem, cost function, and optimisation class
-problem = pybop.Problem(model, parameters, dataset)
+problem = pybop.FittingProblem(parameters, dataset, model)
```

Comment:
I think model could be the first parameter in the inputs, correct? This would allow us to avoid a breaking change here, and match user expectations coming from PyBaMM.